analytics/src/main/java/com/segment/analytics/internal/Utils.java

```diff
@@ -308,8 +313,21 @@ public static String getDeviceId(Context context) {
 
     /** Returns a shared preferences for storing any library preferences. */
     public static SharedPreferences getSegmentSharedPreferences(Context context, String tag) {
-        return obtainDeviceProtectedStorageContext(context)
-                .getSharedPreferences("analytics-android-" + tag, MODE_PRIVATE);
+        Context storageContext;
+        String prefsName = "analytics-android-" + tag;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            // All N devices have split storage areas, but we may need to recreate the preferences in the new device
+            // protected storage area, which is where the data lives from now on.
+            final Context deviceContext = context.createDeviceProtectedStorageContext();
+            if (!deviceContext.moveSharedPreferencesFrom(context,
```

Comment:
Not sure if it would work properly...
I see two pretty unpleasant use cases here:
1. The device [supports FBE](https://source.android.com/security/encryption/file-based) and the device is in the `locked` state (i.e. the boot is just happened and the user hasn't used the credentials to unlock device). Then `moveSharedPreferencesFrom` would return `false` since the user credential storage is not accessible (yet)
2. The device is in the `unlocked` state (i.e. the user used the credentials to unlock device). In that case `moveSharedPreferencesFrom` would always (unconditionally) overwrite what we have in the device storage at the moment.